src/Sarif.Converters/ClangTidyConverter.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Converters.ClangTidyObjectModel;
+
+using YamlDotNet.RepresentationModel;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class ClangTidyConverter : ToolFileConverterBase
+    {
+        private const string ToolInformationUri = "https://clang.llvm.org/extra/clang-tidy/";
+
+        public override string ToolName => "Clang-Tidy";
+
+        public override void Convert(Stream input, IResultLogWriter output, OptionallyEmittedData dataToInsert)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            var yaml = new YamlStream();
+            IDeserializer deserializer = new DeserializerBuilder().Build();
+            using var textReader = new StreamReader(input);
+            ClangTidyLog order = deserializer.Deserialize<ClangTidyLog>(textReader);
+
+            var results = new List<Result>();
+            foreach (ClangTidyDiagnostic entry in order.Diagnostics)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var run = new Run
+            {
+                Tool = new Tool
+                {
+                    Driver = new ToolComponent
+                    {
+                        Name = ToolName,
+                        InformationUri = new Uri(ToolInformationUri),
+                    }
+                },
+                Results = results,
+            };
+
+            PersistResults(output, results, run);
+        }
+
+        internal Result CreateResult(ClangTidyDiagnostic entry)
+        {
+            entry = entry ?? throw new ArgumentNullException(nameof(entry));
+
+            Result result = new Result()
+            {
+                RuleId = entry.DiagnosticName,
```

Comment:
>entry.DiagnosticName [](http://example.com/codeflow?start=25&length=20)

from this, you can create the ReportingDescriptor

Reply:
each rule can point to its own documentation: [https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-trailing-return-type.html](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-trailing-return-type.html)

change modernize-use-trailing-return-type to another ruleid and it will work

Reply:
added the rules.